src/main/java/com/cburch/logisim/fpga/hdlgenerator/Hdl.java

```diff
@@ -112,6 +167,67 @@ public static String xorOperator() {
     return isVhdl() ? Vhdl.getVhdlKeyword(" XOR ") : "^";
   }
 
+  public static String addOperator(String signalOne, String signalTwo, boolean signed) {
+    return (isVhdl() ? "std_logic_vector(" : "")
+            + typecast(signalOne, signed)
+            + " + "
+            + typecast(signalTwo, signed)
+            + (isVhdl() ? ")" : "");
+  }
+
+  public static String subOperator(String signalOne, String signalTwo, boolean signed) {
+    return (isVhdl() ? "std_logic_vector(" : "")
+            + typecast(signalOne, signed)
+            + " - "
+            + typecast(signalTwo, signed)
+            + (isVhdl() ? ")" : "");
+  }
+
+  public static String shiftlOperator(String signal, int width, int distance, boolean arithmetic) {
+    if (distance == 0) return signal;
+    return isVhdl() ? LineBuffer.formatHdl("{{1}}{{2}} & {{4}}{{3}}{{4}}", signal, splitVector(width - 1 - distance, 0), "0".repeat(distance), distance == 1 ? "'" : "\"")
+                    : LineBuffer.formatHdl("{{{1}}{{2}},{{{3}}{1'b0}}}", signal, splitVector(width - 1 - distance, 0), distance);
+  }
+
+  public static String shiftrOperator(String signal, int width, int distance, boolean arithmetic) {
+    if (distance == 0) return signal;
+    if (arithmetic) {
+      return isVhdl()
+        ? LineBuffer.formatHdl("({{1}}{{2}}0 => {{3}}({{1}})) & {{3}}{{4}}", width - 1, vectorLoopId(), signal, splitVector(width - 1, width - distance))
+        : LineBuffer.formatHdl("{ {{{1}}{{{2}}[{{1}}-1]}},{{2}}{{3}}}", width, signal, splitVector(width - 1, width - distance));
+    } else {
+      return isVhdl()
+        ? LineBuffer.formatHdl("{{1}}{{2}}{{1}} & {{3}}{{4}", distance == 1 ? "'" : "\"", "0".repeat(distance), signal, splitVector(width - 1, width - distance))
+        : LineBuffer.formatHdl("{ {{{1}}{1'b0}},{{2}}{{3}}}", width, signal, splitVector(width - 1, width - distance));
+    }
+  }
+
+  public static String sllOperator(String signal, int width, int distance) {
+    return shiftlOperator(signal, width, distance, false);
+  }
+
+  public static String slaOperator(String signal, int width, int distance) {
+    return shiftlOperator(signal, width, distance, true);
+  }
+
+  public static String srlOperator(String signal, int width, int distance) {
+    return shiftrOperator(signal, width, distance, false);
+  }
+
+  public static String sraOperator(String signal, int width, int distance) {
+    return shiftrOperator(signal, width, distance, true);
+  }
+
+  public static String rolOperator(String signal, int width, int distance) {
+    return isVhdl() ? LineBuffer.formatHdl("{{1}}{{2}} & {{1}}{{3}}", signal, splitVector(width - 1 - distance, 0), splitVector(width - 1, width - distance))
```

Comment:
As the only thing that changes here is formatting string, and LineBuffer calls arguments are the same in both cases, can you refactor and separate setting the formatting string and using `LineBuffer` with arguments next?